.circleci/config.yml

```diff
@@ -0,0 +1,71 @@
+version: 2.1
+
+executors:
+  mac-executor:
+    macos:
+      xcode: "11.2.1"
+    working_directory: ~/code
+
+jobs:
+  install_dependendies:
+    executor: mac-executor
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - gem-cache-v1-{{ checksum "Gemfile.lock" }}
+            - gem-cache-v1
+      - run:
+          name: Bundler
+          command: |
+            bundle install --path vendor/bundle --jobs 4 --retry 3
+      - save_cache:
+          key: gem-cache-v1-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+      - persist_to_workspace:
```

Comment:
This persists the whole workspace right? I don't think we really need that. Is it possible to not store the whole workspace but rather to restore the cache we save in this step in the other jobs?

Reply:
Workspaces are another way of caching and the preferred way when passing data between jobs in a workflow [0]. You are right as long as we require data from package dependency managers only, caching is sufficient. Though, as soon as we need e.g. build output for further processing workspaces become a must. An example is displaying tests that have failed right within the PR [1]. You would not cache intermediate outputs using the caching mechanism but pass it in the workspace.

[0] https://circleci.com/docs/2.0/caching/
[1] https://github.com/orta/danger-junit

Reply:
> An example is displaying tests that have failed right within the PR [1].

Yes but then we would explicitly persist the files we need not the whole repository. I'm not against persisting exactly what we need, but persisting the whole working directory (`.` means we persist _everything_ right?) just because we might need some files in it sometime in the future feels a bit imprecise.

Would the following work:
- Cache and restore dependencies between jobs
- Once we need to persist files between jobs, which are not a fit for the cache use `persist_to_workspace`. An example would be to run Sonarqube in a separate job after the tests and persist just the test report in the work space so that Sonarqube can use it for the coverage calculation

Reply:
Yes would absolutely work. Also yes you would only persist the relevant folder/files to the workspace. `.` is indeed too much here. Will do the change.